docs/reference/search/request/preference.asciidoc

```diff
@@ -1,56 +1,81 @@
 [[search-request-preference]]
 === Preference
 
-Controls a `preference` of which shard copies on which to execute the
-search. By default, the operation is randomized among the available shard 
-copies, unless allocation awareness is used.
+Controls a `preference` of which shard copies on which to execute the search.
+By default, Elasticsearch selects from the available shard copies in an
+unspecified order, taking the <<allocation-awareness,allocation awareness>> and
+<<search-adaptive-replica,adaptive replica selection>> configuration into
+account. However, it may sometimes be desirable to try and route certain
+searches to certain sets of shard copies, for instance to make better use of
+per-copy caches.
+
+Preferences do not _guarantee_ that any particular shard copies are used in a
```

Comment:
we need to be more careful here. Some preferences are guaranteed (like `_only_local`). These are useful for debugging and not production use.

Reply:
Really it's just `_only_local` (undocumented prior to this change, and I wrote this paragraph before adding that :grin:) I reworded this in cae22252f6d and moved it below the description of the options where it flowed better.